### ISSUE TITLE
Can not compile. Recenet change in code??

### ISSUE THREAD

**Issue by peteihis** [ArtOfIllusion/ArtOfIllusion#306]
Hi.

I just pulled the master and started to work on a small fix to something that came up over the summer. Now the build was interrupted because of this:

~~~~text
    [javac] C:\Users\Petri\GH\ArtOfIllusion grid_timing\ArtOfIllusion\src\artofillusion\PluginRegistry.java:55: error: cannot find symbol
    [javac]     categoryClasses.getOrDefault(Plugin.class, List.of()).forEach(plugin -> {
    [javac]                                                    ^
    [javac]   symbol:   method of()
    [javac]   location: interface List
~~~~

Has this been edited lately? The date on he file seems to be yesterday.

**Comment by makiam** [comment]
List.of introduced with JDK9 but AOI build script is still targeted on java 1.8 (see https://github.com/ArtOfIllusion/ArtOfIllusion/blob/master/build.properties )

**Comment by makiam** [comment]
List.of() may be replaced with Collections.emptyList()

**Comment by Sailsman63** [comment]
Yeah, current HEAD requires JDK9 or JDK11. That's where [this discussion](https://sourceforge.net/p/aoi/discussion/47784/thread/c088ce4187/) started.

**Comment by peteihis** [comment]
And I don't seem to remember that we actually had a decision...

I did notice that on a different computer, with a newer Java the piece I wrote at least started up faster, but that is partially thanks to the newer HW. One reason, why I have been holding on to J8 is, that it is the last generartion of Oracle Java that did not require registration. (Got OJ8 on the table top and the AJ8 on laptop). At work I have seomething more recent.

Could somebody check the statistics, how much J8s are still active? I remember that with J6 that was done, but I no longer have any idea how. If the generation 8 has no longer significant use base, then I think it whould be OK to move on.

**Comment by makiam** [comment]
>> More than 56% of applications are now using Java 11 in production (up from [48% in 2022] Java 8 is a close second with nearly 33% of applications

**Comment by peteihis** [comment]
The world seemed to pretty much skip 9 and 10... 

But I think the 33% speaks for it self. I'd say, let's keep 8 with us for a couple of more years and then take the step to 11.

**Comment by Sailsman63** [comment]
>The world seemed to pretty much skip 9 and 10...

They were the first two releases in the new release policy. Only certain releases get Long Term Support. 8 is LTS, as is 11 and 17. The in-betweens do *not* have LTS, and were intended to allow quicker updates for organizations that need *some* stability, but are willing to do platform updates every year or so.

Other than certain organizations that prefer to have paid support contracts, most of the world has moved away from Oracle java, even though Oracle engineers are still heavily involved in java development.

@makiam, where did those statistics come from?

**Comment by makiam** [comment]
https://newrelic.com/resources/report/2023-state-of-the-java-ecosystem

**Comment by makiam** [comment]
This is the first link I googled for 'java ecosystem report'

**Comment by peteihis** [comment]
Closed by #315.
